### PR TITLE
add update_and_fetch_row & update_and_emulate_row

### PR DESCRIPTION
**Pull request by mamimumemoomoo**
Fix: https://github.com/karupanerura/Aniki/issues/15

Please review PR

### CODE REVIEW

File: lib/Aniki.pm [hunk 1]
```diff
@@ -256,6 +256,50 @@ sub update {
...
+        $self->update($_[0], $_[1]);
+
+        my $row = $self->select($_[0]->table_name, $self->_where_row_cond($_[0]->table, $_[0]->row_data), { limit => 1, suppress_result_objects => 1 })->[0];
+        return $row if $self->suppress_row_objects;
+
+        $row->is_new(1);
+        return $row;
+    }
+    else {
+        croak '(Aniki#update_and_fetch_row) condition must be a Aniki::Row object.';
```
Comment (karupanerura) [inline lib/Aniki.pm]: It should check at first of the method for less nest of code.
`croak '...' if blessed $_[0] && $_[0]->isa('Aniki::Row');`

File: lib/Aniki.pm [hunk 2]
```diff
@@ -256,6 +256,50 @@ sub update {
...
+sub update_and_fetch_row {
+    my $self = shift;
+
+    if (blessed $_[0] && $_[0]->isa('Aniki::Row')) {
+        local $Carp::CarpLevel = $Carp::CarpLevel + 1;
+
+        $self->update($_[0], $_[1]);
+
+        my $row = $self->select($_[0]->table_name, $self->_where_row_cond($_[0]->table, $_[0]->row_data), { limit => 1, suppress_result_objects => 1 })->[0];
+        return $row if $self->suppress_row_objects;
```
Comment (karupanerura) [inline lib/Aniki.pm]: Please read doc about `suppress_row_objects`: https://metacpan.org/pod/Aniki#suppress_row_objects-:-Bool

File: lib/Aniki.pm [hunk 3]
```diff
@@ -256,6 +256,50 @@ sub update {
...
 
+sub update_and_fetch_row {
+    my $self = shift;
+
+    if (blessed $_[0] && $_[0]->isa('Aniki::Row')) {
+        local $Carp::CarpLevel = $Carp::CarpLevel + 1;
+
+        $self->update($_[0], $_[1]);
+
+        my $row = $self->select($_[0]->table_name, $self->_where_row_cond($_[0]->table, $_[0]->row_data), { limit => 1, suppress_result_objects => 1 })->[0];
```
Comment (karupanerura) [inline lib/Aniki.pm]: It should follow changing value of primary key or it fetch other row.

File: lib/Aniki.pm [hunk 4]
```diff
@@ -256,6 +256,50 @@ sub update {
     }
 }
 
+sub update_and_fetch_row {
+    my $self = shift;
```
Comment (karupanerura) [inline lib/Aniki.pm]: This method's arguments should be assign to obvious variable.

File: lib/Aniki.pm [hunk 5]
```diff
@@ -256,6 +256,50 @@ sub update {
...
+        $row->is_new(1);
+        return $row;
+    }
+    else {
+        croak '(Aniki#update_and_fetch_row) condition must be a Aniki::Row object.';
+    }
+}
+
+sub update_and_emulate_row {
+    my $self = shift;
```
Comment (karupanerura) [inline lib/Aniki.pm]: ditto

File: lib/Aniki.pm [hunk 6]
```diff
@@ -256,6 +256,50 @@ sub update {
...
+
+        return $row if $self->suppress_row_objects;
+        return $self->guess_row_class($_[0]->table_name)->new(
+            table_name => $_[0]->table_name,
+            handler    => $self,
+            row_data   => $row,
+        );
+    }
+    else {
+        croak '(Aniki#update_and_emulate_row) condition must be a Aniki::Row object.';
```
Comment (karupanerura) [inline lib/Aniki.pm]: ditto

**Review by karupanerura — CHANGES_REQUESTED** [review]
Thank you for your patch. Good job! but, I found some problems of the code.
Please consider these points of view and please fix this patch.

File: lib/Aniki.pm (hunk 1 above)
Comment (mihyaeru21) [reply lib/Aniki.pm]: `croak '...' unless blessed $_[0] && $_[0]->isa('Aniki::Row');` ?

Comment (karupanerura) [reply lib/Aniki.pm]: oops! that's right! thanks!

File: lib/Aniki.pm (hunk 4 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: fix d4c6564fe100f5a9638cfae9597eec7da87c8969

File: lib/Aniki.pm (hunk 5 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: d4c6564fe100f5a9638cfae9597eec7da87c8969

File: lib/Aniki.pm (hunk 1 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: fix a6b8df1f9da2e39735f0e7351e0cafa57814b5c6

File: lib/Aniki.pm (hunk 6 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: a6b8df1f9da2e39735f0e7351e0cafa57814b5c6

File: lib/Aniki.pm (hunk 2 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: if suppress_row_objects is true, can not pass $row object.
Is my understanding correct?
64b13c451ff651bd9dc8cb9c1d0f7ce932f90914

File: lib/Aniki.pm (hunk 3 above)
Comment (mamimumemoomoo) [reply lib/Aniki.pm]: oopus! 8ed502635921f6c1902bc17b0082ca09273957cf

File: lib/Aniki.pm (hunk 2 above)
Comment (karupanerura) [reply lib/Aniki.pm]: Yes! That's right. so this is meaningless branch.

File: lib/Aniki.pm [hunk 7]
```diff
@@ -256,6 +256,44 @@ sub update {
...
+    croak '(Aniki#update_and_emulate_row) condition must be a Aniki::Row object.' unless blessed $row && $row->isa('Aniki::Row');
+
+    local $Carp::CarpLevel = $Carp::CarpLevel + 1;
+
+    $self->update($row, $set);
+
+    my $row_data = $row->get_columns;
+    $row_data->{$_} = $set->{$_} for keys %$set;
+
+    $row_data = $self->filter_on_update($row->table_name, $row_data);
```
Comment (karupanerura) [inline lib/Aniki.pm:288]: please follow `suppress_row_objects` at here.

**Review by karupanerura — CHANGES_REQUESTED** [review]

**Comment by karupanerura** [comment]
@mamimumemoomoo Thank you. Please fix a point and I'll merge it.

**Comment by karupanerura** [comment]
OK. I'll fix it. Good Job!

**Review by karupanerura — APPROVED** [review]

**Comment by karupanerura** [comment]
@mamimumemoomoo Hi.
I've released this feature as 1.04_02 on CPAN (developer release).
Would you try it in your application?

**Comment by mamimumemoomoo** [comment]
I'm sorry to reply late.
Thank you for support and review.
I'll try new version.

**Comment by karupanerura** [comment]
@mamimumemoomoo How is the progress?
If there are no anxiety, I'll release it as new version.

**Comment by mamimumemoomoo** [comment]
@karupanerura 
Thank you! We're on track for now.
I'm looking forward to it.

**Comment by karupanerura** [comment]
released it as 1.05

**Pull request merged**